src/sentry/sentry_metrics/metrics_interface.py

```diff
@@ -33,9 +35,10 @@ def set(
         project_id: int,
         metric_name: str,
         value: Sequence[int],
-        unit: Optional[str],
         tags: Mapping[str, str],
-    ) -> None:
+        unit: Optional[str] = None,
+        retention_days: Optional[int] = 90,
```

Comment:
Not sure if the default should be 90 here... can we make it `None` instead? Snuba applies the right value in https://github.com/getsentry/snuba/blob/884843d9c10cb6f21d90aff8954d7490779df577/snuba/settings/__init__.py#L205, and this can be overridden by the user via settings. I think it's better to never hardcode 90 here.

Reply:
changed 90 -> None

Reply:
As I was putting in the schema validation, I noticed that `retention_days` is required: https://github.com/getsentry/sentry-kafka-schemas/blob/main/schemas/ingest-metrics.v1.schema.json#L77. I think I might need to set a default value in that case.

Reply:
If Lyn is very concerned with a default causing problems here then I'm fine with making it optional in the schema (though we'll need to make sure that the indexer consumer handles this case correctly first). 

If Lyn isn't that concerned I think 90 days is the right default.

Reply:
Sure, and if we decide a default is okay, I can also mention in the docstring of the methods that a 90day default is being applied, just so the user is aware.

Reply:
It's about us respecting the retention that is configured for every organization (which is not necessarily the same - for example many orgs are on the 30 day retention option), and self hosted can also configure other values via the settings file.

With this API, we're either:
a) putting the responsibility on the caller to look up the configured retention for each org every time one of these metrics functions is called then pass it through here, which can be cumbersome and prone to error, or:
b) applying 90 days for all users even if 30 was actually configured

I think it'd be better to try and make it less prone to error and not have the caller of this method even have to calculate the retention days by themselves, and also apply the actual org configured retention rather hardcoding them all to 90.

Reply:
I don't know that org retention is universally applicable here. I can imagine usecases for metrics where there's no customer data stored (eg timeseries per issue, where we would want to store data way longer). But I hadn't considered that point before and respecting org retention is certainly an interesting default, probably a better one than 90d.

Reply:
So, I think in order to get the org-configured retention, we would need to make DB calls for each metric being emitted, and I think this has been attempted before and caused issues. Another point is that, for the Relay-based implementation of this backend, we wouldn't need to get the org-configured retention in the interface since Relay already takes care of that before sending metrics over to the indexer. 

How do we feel about changing the indexer schema to make the retention optional?

Reply:
we definitely can't look up retention from the indexer in SaaS. that has caused an incident

Reply:
@onewland I believe it should be possible using batched memcache/redis caching. I don't know that we'd need this to be strongly consistent with the actual db

Reply:
I guess my thought process is, if there's no strong reason behind why `retention_days` is required in the schema right now, couldn't we just make it optional?

What that would look like here:
The indexer schema would make `retention_days` optional, the default in all of the metrics backends would be `retention_days=None`, and then Snuba later populates it as it does already if the user didn't provide an override in the metrics backend.

Reply:
I don't think Snuba has this information at all. Where do you see it populating retention?

Reply:
I was thinking about this https://github.com/getsentry/snuba/blob/master/snuba/datasets/events_format.py#L103-L115, which has a set of valid retention days that Snuba checks against (and if the `retention_days` passed in isn't valid, applies one of the valid values). I guess this doesn't help here though, since we want to not put the responsibility on the caller to find the right retention.